### ISSUE TITLE
No data being streamed on topics

### ISSUE THREAD

**Issue by ghost** [mje-nz/natnet_ros#11]
**Environment**
Motive version: 2.1.1
ROS distribution: Melodic
Operating system: Ubuntu 18.04.5
Python version: 2.7.17

**`natnet_ros` version**
`natnet_ros` branch: master
`natnet_ros` commit: a859ada

First of all: Thank you for creating this package which helped me a lot already.
This is not a bug report, in fact there is not even an error message. When I run 
```
natnet_ros client _server:=172.16.0.3
```
a connection between the ROS PC and the Motive PC seems to be correctly established, because all the topics are listed:
```
rostopic list
/mocap/markers/leftovers
/mocap/markers/vis
/mocap/rigid_bodies/ball/marker1
/mocap/rigid_bodies/ball/marker2
/mocap/rigid_bodies/ball/marker3
/mocap/rigid_bodies/ball/marker4
/mocap/rigid_bodies/ball/marker5
/mocap/rigid_bodies/ball/markers
/mocap/rigid_bodies/ball/pose
/rosout
/rosout_agg
/tf
```
However, no motive data is being published on the topics, i.e. 
```
rostopic echo /mocap/rigid_bodies/ball/pose 
```
returns nothing. The same is true for all the other topics. Since the ball is correctly tracked in Motive, I assume that the problem has to be somewhere in the data streaming. Unfortunately, I have no idea where to search the error, because obviously the streaming of the model definitions works fine. Otherwise, the topics would not be generated.
I would be very glad for any advice you could give me. Thanks in advance!

**Comment by mje-nz** [comment]
Thanks for the report, could you post a screenshot of your streaming settings in Motive?

**Comment by ghost** [comment]
Thanks for the quick response, here are the settings:
![Unbenannt](https://user-images.githubusercontent.com/78913851/107766434-befddd00-6d33-11eb-9981-50d5133e6dff.PNG)

**Comment by mje-nz** [comment]
From memory I think you need “labelled markers” on as well or something,
let me know if any of those fix it.

On 13/02/2021, at 1:11 AM, ralfroemer <notifications@github.com> wrote:

﻿

Thanks for the quick response, here are the settings:
[image: Unbenannt]
<https://user-images.githubusercontent.com/78913851/107766434-befddd00-6d33-11eb-9981-50d5133e6dff.PNG>

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/mje-nz/natnet_ros/issues/11#issuecomment-778158882>, or
unsubscribe

**Comment by ghost** [comment]
Unfortunately, that as well as changing the other options in the streaming tab did not solve the issue.
However, this warning might be worth mentioning:
![Bildschirmfoto vom 2021-02-15 14-49-53](https://user-images.githubusercontent.com/78913851/107954694-15218900-6f9d-11eb-9260-3e292d7f4385.png)
Could these warnings be related to the problem? I am still confused about how the correct topics are being generated but no data is being streamed onto them.

**Comment by aarsht7** [comment]
@ralfroemer  were you able to solve this error?

**Comment by MaxorPaxor** [comment]
Has anyone solved this?
@aarsht7 @ralfroemer

**Comment by aarsht7** [comment]
Nope, but I have created a new package with more features for ros1 if would you like to use for
Ros1 (https://github.com/aarsht7/natnet_ros_cpp)

For ros2 you can follow this package but its WIP
Ros2 (https://github.com/MOCAP4ROS2-Project/mocap4ros2_optitrack)

@MaxorPaxor

**Comment by MaxorPaxor** [comment]
> Nope, but I have created a new package with more features for ros1 if would you like to use for Ros1 (https://github.com/aarsht7/natnet_ros_cpp)
> 
> For ros2 you can follow this package but its WIP Ros2 (https://github.com/MOCAP4ROS2-Project/mocap4ros2_optitrack)
> 
> @MaxorPaxor

Thank you for your reply, will try it
